client/src/store/contest.js

```diff
@@ -0,0 +1,103 @@
+import Vue from 'vue';
+import * as api from '@/api/contest';
+
+const contestState = {
+  accumulatedAnsweredQuestions: {
+    data: {},
+    lastUpdateDateTime: null,
+  },
+};
+
+const getters = {
+  accumulatedAnsweredQuestions: state => {
+    const data = state.accumulatedAnsweredQuestions.data;
+    return Object.keys(data).map(key => {
+      return {
+        label: key,
+        borderColor: data[key].color,
+        data: data[key].answers,
+        totalScore: data[key].answers[data[key].answers.length - 1].y,
+      };
+    });
+  },
+  lastUpdateDateTime: state => state.accumulatedAnsweredQuestions.lastUpdateDateTime,
+};
+
+function getRandomColor() {
+  var letters = '0123456789ABCDEF'.split('');
+  var color = '#';
+  for (var i = 0; i < 6; i++) {
+    color += letters[Math.floor(Math.random() * 16)];
+  }
+  return color;
+}
+
+const setAnsweredQuestionsMutations = {
+  setAccumulatedAnsweredQuestions(state, answeredQuestions) {
+    state.accumulatedAnsweredQuestions.data = getUpdatedAnswers(
+      answeredQuestions,
+      state.accumulatedAnsweredQuestions.data,
+    );
+  },
+  setLastUpdateDateTime(state, lastUpdateDateTime) {
+    state.accumulatedAnsweredQuestions.lastUpdateDateTime = lastUpdateDateTime;
+  },
+};
+
+const mutations = {
+  ...setAnsweredQuestionsMutations,
+};
+
+function getUpdatedAnswers(inputData, currData = {}) {
```

Comment:
If it's possible could you please use arrow function? (getUpdatedAnswers, getRandomColor, getUpdatedAnswersByContender)

Reply:
sure